### PR TITLE
Cy 3487 refresh gcp token

### PR DESCRIPTION
**Pull request by AdarShaked**

### CODE REVIEW

File: helm_sdk/utils.py [hunk 1]
```diff
@@ -152,3 +154,10 @@ def prepare_set_parameters(set_values):
             raise CloudifyHelmSDKError(
                 "set parameter name or value is missing")
     return set_list
+
+
+def validate_no_collisions_between_params_and_flags(flags):
+    if [flag for flag in flags if flag['name'] in FLAGS_LIST_TO_VALIDATE]:
+        raise CloudifyHelmSDKError(
+            'Please do not pass {flags_list} flags inside "flags" list'.format(
+                flags_list=FLAGS_LIST_TO_VALIDATE))
```
Comment (EarthmanT) [inline helm_sdk/utils.py:163]: Not clear to me what we should be passing. Maybe a more useful message?

File: setup.py [hunk 2]
```diff
@@ -45,7 +45,9 @@ def get_version(rel_file='plugin.yaml'):
     install_requires=[
         "cloudify-common>=4.5.5",
         "cloudify-utilities-plugins-sdk==0.0.28",  # obfuscate_passwords
-        "pyyaml==3.12"
+        "pyyaml==3.12",
```
Comment (EarthmanT) [inline setup.py:48]: Why should we use specifically 3.12?

File: setup.py [hunk 3]
```diff
@@ -45,7 +45,9 @@ def get_version(rel_file='plugin.yaml'):
...
+        'oauth2client'  # used in GCPServiceAccountAuthentication
+
     ],
     test_requires=[
         "cloudify-common>=4.5.5",
```
Comment (EarthmanT) [inline setup.py:53]: IMO, we don't need to support anything before 5.0.5. But ask Ofer.

File: cloudify_helm/authentication.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+class GCPServiceAccountAuthentication(KubernetesApiAuthentication):
+    PROPERTY_GCP_SERVICE_ACCOUNT = 'gcp_service_account'
+
+    SCOPES = ['https://www.googleapis.com/auth/cloud-platform']
+
+    def _get_token(self):
+        service_account_file_content = self.authentication_data.get(
+            self.PROPERTY_GCP_SERVICE_ACCOUNT
+        )
```
Comment (EarthmanT) [inline cloudify_helm/authentication.py:58]: not impt but you can put this on the line above.
```
self.PROPERTY_GCP_SERVICE_ACCOUNT)
```

File: cloudify_helm/authentication.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            except HelmKuberentesAuthenticationError:
+                self.logger.debug(
+                    'Authentication option {0} cannot be used'.format(
+                        variant.__name__)
+                )
+
+        self.logger.debug(
+            'Cannot generate Bearer token - no suitable authentication '
+            'variant found for {0} properties'.format(self.authentication_data)
+        )
```
Comment (EarthmanT) [inline cloudify_helm/authentication.py:102]: Also not required, but while I'm nit-picking, I think you should use keys for `.format`, i.e. `'{props}'.format(props=self.authentication_data)`.

File: cloudify_helm/configuration.py [hunk 6]
```diff
@@ -104,7 +104,8 @@ class FileContentConfiguration(KubernetesConfiguration):
     def _get_kubeconfig(self):
         if self.FILE_CONTENT_KEY in self.configuration_data:
             file_content = self.configuration_data[self.FILE_CONTENT_KEY]
-            with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
+            with tempfile.NamedTemporaryFile(delete=False,
+                                             mode='w') as tmp_file:
```
Comment (EarthmanT) [inline cloudify_helm/configuration.py:108]: Are we 100% ok with this being deleted after the context closes?

**Review by EarthmanT — APPROVED** [review]

File: setup.py (hunk 2 above)
Comment (AdarShaked) [reply setup.py:48]: All our plugins are using this version. Its because old managers(packages issues)

File: cloudify_helm/configuration.py (hunk 6 above)
Comment (AdarShaked) [reply cloudify_helm/configuration.py:108]: Its not being deleted because of `delete=False`

File: cloudify_helm/authentication.py (hunk 5 above)
Comment (AdarShaked) [reply cloudify_helm/authentication.py:102]: Agree. I will use keywords as standard from now.

File: helm_sdk/utils.py (hunk 1 above)
Comment (AdarShaked) [reply helm_sdk/utils.py:163]: Done.

File: cloudify_helm/authentication.py (hunk 4 above)
Comment (AdarShaked) [reply cloudify_helm/authentication.py:58]: Done

File: setup.py (hunk 3 above)
Comment (AdarShaked) [reply setup.py:53]: Asked. Its good like that for now.

**Review by EarthmanT — APPROVED** [review]

**Pull request merged**